shell/server/proxy.js

```diff
@@ -257,8 +341,22 @@ Meteor.methods({
       if (identityId != apiToken.identityId && identityId != grain.identityId &&
           !ApiTokens.findOne({'owner.user.identityId': identityId, parentToken: hashedToken })) {
         var owner = {user: {identityId: identityId, title: title}};
-        SandstormPermissions.createNewApiToken(globalDb, {rawParentToken: token}, apiToken.grainId,
-                                               apiToken.petname, {allAccess: null}, owner);
+
+        // Create a new API token for the identity redeeming this token.
+        var result = SandstormPermissions.createNewApiToken(
```

Comment:
@dwrensha this hopefully avoids the need to modify the data model for ApiTokens.